src/mpb.cpp

```diff
@@ -173,7 +185,7 @@ complex<double> eigenmode_amplitude(void *vedata, const vec &p,
 #undef D
 
   return (complex<double>(double(real(ret)), double(imag(ret)))
-	  * amp_func(p));
+	  * amp_func(p)) * std::polar(1.0, TWOPI*phase);
```

Comment:
Removing `std::polar(1.0, TWOPI*phase)` from this line produces results which match the expected results from the `amp_func` source. This is shown in the figure below. It does not seem obvious why removing the `phase` yields the correct results but it does. The rest of this PR simply fixes the 1-pixel boundary bug. However, even with this fix, `python/tests/mode_coeffs.py` is still failing: the computed and expected results differ by much more than a few decimal places.

![ez_real_ampfunc_geteigenmode_vary_theta2](https://user-images.githubusercontent.com/7152530/46101300-6f2d1400-c180-11e8-83c6-b67e3386c984.png)